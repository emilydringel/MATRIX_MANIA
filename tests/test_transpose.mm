def void transpose(matrix<int> original, matrix<int> result){
    int sizeOfR1 = getRows(original);
    int sizeOfC1= getColumns(original);

    for(int i = 0; i < sizeOfR1; i=i+1){
	    for(int j = 0; j < sizeOfC1; j=j+1){
            result[i,j] = original[j,i];
	    }
    }
}

def int main(){

    matrix<int> m = [1,2,3; 4,5,6];
    matrix<int> n = [0,0;0,0;0,0];
    matrix<int> tranposed = transpose(m,n);


    printm(m);
    printm(transposed);

    /*printm(transpose([7,8,9,10; 11,12,13,14]);*/
    printm(transpose([7,8,9,10; 11,12,13,14],[0,0;0,0;0,0;0,0]);
} 
