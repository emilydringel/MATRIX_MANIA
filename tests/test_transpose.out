1 2 3
4 5 6
1 4
2 5
3 6

7 8 9 10
11 12 13 14
7 11
8 12
9 13
10 14
